cmd/crossplane/core/core.go

```diff
@@ -96,7 +102,34 @@ type startCommand struct {
 }
 
 // Run core Crossplane controllers.
-func (c *startCommand) Run(s *runtime.Scheme, log logging.Logger) error { //nolint:gocyclo // Only slightly over (11).
+func (c *startCommand) Run(s *runtime.Scheme, log logging.Logger) error { //nolint:gocyclo // Only slightly over.
+	if c.Profile != "" {
+		// NOTE(negz): These log messages attempt to match those emitted by
+		// controller-runtime's metrics HTTP server when it starts.
+		log.Debug("Profiling server is starting to listen", "addr", c.Profile)
+		go func() {
+
+			// Registering these explicitly ensures they're only served by the
+			// HTTP server we start explicitly for profiling.
+			mux := http.NewServeMux()
+			mux.HandleFunc(pprofPath, pprof.Index)
+			mux.HandleFunc(filepath.Join(pprofPath, "cmdline"), pprof.Cmdline)
+			mux.HandleFunc(filepath.Join(pprofPath, "profile"), pprof.Profile)
+			mux.HandleFunc(filepath.Join(pprofPath, "symbol"), pprof.Symbol)
+			mux.HandleFunc(filepath.Join(pprofPath, "trace"), pprof.Trace)
+
+			s := &http.Server{
+				Addr:         c.Profile,
+				ReadTimeout:  2 * time.Minute,
+				WriteTimeout: 2 * time.Minute,
+				Handler:      mux,
+			}
+			log.Debug("Starting server", "type", "pprof", "path", pprofPath, "addr", s.Addr)
+			err := s.ListenAndServe()
+			log.Debug("Profiling server has stopped listening", "error", err)
+		}()
+	}
+
```

Comment:
This is repeated here and in the RBAC manager. I considered adding it to crossplane-runtime as a helper, but so far it's pretty straightforward. I'd prefer to hold off for now, but we can add it there if we start propagating it elsewhere (e.g. to providers).

I also considered adding Helm chart plumbing to enable this, but I also would prefer not to. I see this as mostly a developer tool - not something that needs to be made particularly easy to turn on.